src/pod/utils.ts

```diff
@@ -137,17 +185,27 @@ export function assertPodName(value: unknown): asserts value is string {
 
 /**
  * Converts pods list to bytes array
- *
- * @param list list of pods
  */
-export function podListToBytes(list: Pod[]): Uint8Array {
-  assertPods(list)
+export function podListToBytes(pods: Pod[], sharedPods: SharedPod[]): Uint8Array {
+  assertPods(pods)
+  assertSharedPods(sharedPods)
 
-  if (list.length === 0) {
+  if (pods.length === 0 && sharedPods.length === 0) {
     return new Uint8Array([0])
   }
 
-  return stringToBytes(list.map(pod => `${pod.name},${pod.index}`).join('\n') + '\n')
+  const allPods =
+    [...pods, ...sharedPods]
+      .map(pod => {
+        if (isPod(pod)) {
+          return `${pod.name},${pod.index}`
+        } else if (isSharedPod(pod)) {
+          return `${pod.name},${bytesToHex(pod.address)}`
+        }
+      })
+      .join('\n') + '\n'
```

Comment:
this mixed data-structure is a bit strange for me, I thought one will be in a different array than the other to clearly have distinction between the two.

Reply:
unfortunately, such a structure in fairos. We need to get away from this.